### PR TITLE
`Communication`: Add profile picture to user selector

### PR DESCRIPTION
**Pull request by asliayk**
#### General

- [X] This is a small issue that I tested locally.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [X] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [X] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Currently, when users select individuals to create a direct/group chat, only names are displayed, and profile pictures are not visible. This can make it harder to identify the correct person.

### Description

Added profile picture component to user selector to get relevant profile pictures.

### Steps for Testing

Prerequisites:
- 1 Instructor/Student
- 1 Course with Communication enabled

1. Log in to Artemis
2. Navigate to Communication section of a course
3. Click + button next to the search area to create direct/group chat.
4. Notice that when the user searches for names, the suggestions now contain relevant profile pictures.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

**direct chat dialog**
image

**group chat dialog**
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced user selection display in the dropdown with profile pictures and user labels.
  
- **Bug Fixes**
	- Retained existing functionality for search status messages and user role filters.

- **Tests**
	- Updated test suite to include a mock for the ProfilePictureComponent, ensuring accurate testing of the CourseUsersSelectorComponent and verifying the rendering of profile pictures.

### CODE REVIEW

File: src/main/webapp/app/shared/course-users-selector/course-users-selector.component.html [hunk 1]
```diff
@@ -13,13 +13,30 @@
...
             [inputFormatter]="usersFormatter"
-            [resultFormatter]="usersFormatter"
+            [resultTemplate]="rt"
             [editable]="false"
             [focusFirst]="false"
             (input)="onInputChange($event)"
             placement="bottom-start"
             #instance="ngbTypeahead"
         />
+        <ng-template #rt let-user="result" let-term="term">
```
Comment (PaRangger) [inline src/main/webapp/app/shared/course-users-selector/course-users-selector.component.html:23]: Can you rename this to "#resultTemplate", I was at first a bit confused when i just saw "rt" at the top.

File: src/test/javascript/spec/component/shared/course-users-selector/course-users-selector.component.spec.ts [hunk 2]
```diff
@@ -6,14 +6,15 @@ import { NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
...
-import { MockPipe, MockProvider } from 'ng-mocks';
+import { MockComponent, MockPipe, MockProvider } from 'ng-mocks';
 import { CourseManagementService } from 'app/course/manage/course-management.service';
 import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
 import { CourseUsersSelectorComponent, SearchRoleGroup } from 'app/shared/course-users-selector/course-users-selector.component';
 import { UserPublicInfoDTO } from 'app/core/user/user.model';
 import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
 import { ArtemisSharedModule } from 'app/shared/shared.module';
 import { TranslateModule } from '@ngx-translate/core';
+import { ProfilePictureComponent } from '../../../../../../main/webapp/app/shared/profile-picture/profile-picture.component';
```
Comment (PaRangger) [inline src/test/javascript/spec/component/shared/course-users-selector/course-users-selector.component.spec.ts:17]:
```suggestion
import { ProfilePictureComponent } from 'app/shared/profile-picture/profile-picture.component';
```

**Review by PaRangger — CHANGES_REQUESTED** [review]

**Comment by PaRangger** [comment]
Tested on TS3, looks great! I left some small nitpicks for the code.

**Review by sachmii — APPROVED** [review]
Tested on TS3, looks good.

Bildschirmfoto 2024-11-13 um 10 47 32

**Review by Longppham — COMMENTED** [review]
I tested the student side on TS4. After following the described steps for testing, everything worked as described! Showed multiple pictures and names but also for 1 search result!

Since this works very well and having pictures added makes searching a lot easier. I would love to see this implemented.



image
image

**Comment by Longppham** [comment]
I tested the student side on TS4. After having the review request, I tried out the same stuff and it workes as described! I attach the screenshots here. Worked again for one single one as well as for multiple search results!

image
image

**Review by PaRangger — APPROVED** [review]
Code LGTM, Tested on TS3

**Comment by rabeatwork** [comment]
Bit picky, but could you adjust the alignment to center the picture and text?
Screenshot 2024-11-13 at 13 49 16

**Review by vinceclifford — APPROVED** [review]
Tested on TS5, works as expected. Same nitpick as Ramona.

**Review by sachmii — APPROVED** [review]
re-approve

**Review by HawKhiem — APPROVED** [review]
Tested on TS6. Profiles pictures showed up as described

image

**Review by ahbitaqu — APPROVED** [review]
[Tested on TS1]
No issues found here. Very welcome feature!

**Review by Malekos74 — APPROVED** [review]
### Server tested on: 01
Works as expected. I like it

**Review by sawys777 — APPROVED** [review]
Tested on TS1, everything worked as expected.

**Review by krusche — APPROVED** [review]

**Pull request merged**